src/validators/tasks.py

```diff
@@ -78,28 +77,21 @@ async def register_validators(keystores: Keystores, deposit_data: DepositData) -
         validators=validators,
     )
     registry_root = None
-    deadline = None
     while True:
         latest_registry_root = await validators_registry_contract.get_registry_root()
-        latest_deadline = datetime.now(timezone.utc) + timedelta(
-            seconds=oracles.signature_validity_period
-        )
-        if (
-            not registry_root
-            or registry_root != latest_registry_root
-            or deadline <= datetime.now(timezone.utc)
-        ):
+
+        deadline = get_current_timestamp() + oracles.signature_validity_period
+        if not registry_root or registry_root != latest_registry_root:
```

Comment:
looks like bug. `oracles_request` will be cached with old deadline while registry root doesn't change